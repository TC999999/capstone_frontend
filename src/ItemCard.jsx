import { Link } from "react-router-dom";
// import { useState, useEffect, useContext } from "react";
// import UserContext from "./UserContext";

const ItemCard = ({ item }) => {
  //   const { user } = useContext(UserContext);
  return (
    <div className="item-card">
      {item.id ? (
        <h3>
          <Link to={`/items/${item.id || item.itemID}`}>{item.name}</Link>
        </h3>
      ) : (
        <h3>
          <Link to={`/items/${item.itemID}`}>{item.name}</Link>
        </h3>
      )}

      <p>
        <b>Price:</b> {item.initialPrice}
      </p>
      {item.imageURL && (
        <img width="200" height="200" src={`${item.imageURL}`}></img>
      )}
      {item.sellerUser && (
        <p>
          <b>Seller:</b>{" "}
          <Link to={`/users/${item.sellerUser}`}>{item.sellerUser}</Link>
        </p>
      )}
      <p>
        <b>Condition:</b> {item.condition}
      </p>
      <p>
        <b>Description:</b> {item.description}
      </p>
      {item.location && (
        <p>
          <b>Location:</b> {item.location.city}, {item.location.regionOrState}
        </p>
      )}
    </div>
  );
};

export default ItemCard;
