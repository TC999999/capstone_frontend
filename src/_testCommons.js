const TEST_USER = {
  username: "u1",
  firstName: "fn1",
  lastName: "ln1",
  email: "testemail1@gmail.com",
  address: "100 address",
  zipCode: "11111",
  regionOrState: "state1",
  city: "city1",
  country: "country1",
  latitude: "10.99876",
  longitude: "-99.99981",
  isAdmin: false,
  isFlagged: false,
  items: [
    {
      itemID: 1,
      name: "i1",
      initialPrice: 100,
      imageURL: "test_url_1",
      condition: "great",
      description: "test item",
    },
    {
      itemID: 2,
      name: "i2",
      initialPrice: 100,
      imageURL: "test_url_2",
      condition: "great",
      description: "test item",
    },
  ],
  reviews: [],
  reports: [],
  pastPurchases: [
    {
      purchaseID: 1,
      itemName: "i4",
      sellerUser: "u3",
      imageURL: "test_url_4",
      price: 150,
      soldAt: expect.any(Date),
      fromFlaggedUser: false,
    },
  ],
};

const TEST_ITEM = {
  id: 1,
  name: "i1",
  imageURL: "test_url_1",
  initialPrice: 100,
  condition: "great",
  description: "test item",
  sellerUser: "u1",
  isSold: false,
  types: [{ id: 1, name: "electronics" }],
  location: {
    city: "city1",
    address: "100 address",
    latitude: "10.99876",
    longitude: "-99.99981",
    regionOrState: "state1",
    country: "country1",
    zipCode: "11111",
  },
};

const TEST_ITEMS = [
  {
    id: 3,
    name: "i3",
    imageURL: "test_url_3",
    initialPrice: 100,
    condition: "great",
    description: "test item",
    sellerUser: "u2",
    types: [
      { id: 1, name: "electronics" },
      { id: 3, name: "books" },
    ],
    location: {
      city: "city2",
      regionOrState: "state2",
      country: "country2",
      zipCode: "22222",
    },
  },
  {
    id: 2,
    name: "i2",
    imageURL: "test_url_2",
    initialPrice: 100,
    condition: "great",
    description: "test item",
    sellerUser: "u1",
    types: [
      { id: 1, name: "electronics" },
      { id: 2, name: "movies" },
    ],
    location: {
      city: "city1",
      regionOrState: "state1",
      country: "country1",
      zipCode: "11111",
    },
  },
  {
    id: 1,
    name: "i1",
    imageURL: "test_url_1",
    initialPrice: 100,
    condition: "great",
    description: "test item",
    sellerUser: "u1",
    types: [{ id: 1, name: "electronics" }],
    location: {
      city: "city1",
      regionOrState: "state1",
      country: "country1",
      zipCode: "11111",
    },
  },
];

const TEST_MESSAGE = {
  id: 3,
  from: "u2",
  to: "u1",
  itemID: 1,
  itemName: "i3",
  body: "test body",
  createdAt: "2024-07-09 14:23:43.469126-04",
};

const TEST_MESSAGE_2 = {
  id: 3,
  from_username: "u2",
  to_username: "u1",
  itemID: 1,
  item_name: "i3",
  body: "test body",
  sent_at: "2024-07-09 14:23:43.469126-04",
};

const TEST_MESSAGES = {
  conversation: [
    {
      id: 2,
      from_username: "u3",
      to_username: "u1",
      item_name: "i4",
      itemID: 4,
      body: "test body",
      sent_at: "2024-07-09 14:23:43.469126-04",
    },
    {
      id: 1,
      from_username: "u1",
      to_username: "u3",
      item_name: "i4",
      itemID: 4,
      body: "test body",
      sent_at: "2024-07-09 14:23:43.469126-04",
    },
  ],
};

const TEST_REVIEW = {
  id: 1,
  reviewed: "u3",
  reviewer: "u1",
  rating: 10,
  body: "test review",
  madeAt: "2024-07-09 14:23:43.469126-04",
};

const TEST_REPORT = {
  id: 1,
  reportedUser: "u2",
  reporterUser: "u1",
  body: "test body",
  madeAt: "2024-07-09 14:23:43.469126-04",
  isCleared: false,
};

const TEST_REPORTS = [
  {
    id: 1,
    reportedUser: "u2",
    reporterUser: "u1",
    body: "test body",
    madeAt: "2024-07-09 14:23:43.469126-04",
    isCleared: false,
  },
  {
    id: 2,
    reportedUser: "u1",
    reporterUser: "u2",
    body: "test body",
    madeAt: "2024-07-09 14:23:43.469126-04",
    isCleared: false,
  },
];

const TEST_PURCHASE = {
  purchaseID: 1,
  itemName: "i4",
  sellerUser: "u3",
  imageURL: "test_url_4",
  price: 150,
  soldAt: "2024-07-09 14:23:43.469126-04",
  fromFlaggedUser: false,
};

module.exports = {
  TEST_USER,
  TEST_ITEM,
  TEST_ITEMS,
  TEST_MESSAGE,
  TEST_MESSAGE_2,
  TEST_MESSAGES,
  TEST_REPORT,
  TEST_REPORTS,
  TEST_REVIEW,
  TEST_PURCHASE,
};
