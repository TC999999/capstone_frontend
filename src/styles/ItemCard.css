.item-card {
  display: flex;
  justify-content: flex-start;
  border: solid 1px blue;
  box-shadow: 5px 5px 5px black;
  margin: 10px;
  background-color: rgb(228, 220, 220);
}

.item-card a {
  text-decoration-line: none;
}

.item-card h3 {
  font-family: Arial, Helvetica, sans-serif;
}

.item-info {
  margin: 5px;
}

.additional-item-info {
  padding-top: 45px;
}
