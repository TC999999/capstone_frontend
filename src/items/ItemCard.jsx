import { Link } from "react-router-dom";
import "../styles/ItemCard.css";

const ItemCard = ({ item }) => {
  return (
    <div className="item-card">
      <div className="item-image-name item-info">
        {item.id ? (
          <h3>
            <Link to={`/items/${item.id || item.itemID}`}>{item.name}</Link>
          </h3>
        ) : (
          <h3>
            <Link to={`/items/${item.itemID}`}>{item.name}</Link>
          </h3>
        )}

        {item.imageURL && (
          <img width="200" height="200" src={`${item.imageURL}`}></img>
        )}
      </div>
      <div className="additional-item-info item-info">
        <p>
          <b>Price:</b> ${item.initialPrice}
        </p>
        {item.sellerUser && (
          <p>
            <b>Seller:</b>{" "}
            <Link to={`/users/${item.sellerUser}`}>{item.sellerUser}</Link>
          </p>
        )}
        <p>
          <b>Condition:</b> {item.condition}
        </p>
        <p>
          <b>Description:</b> {item.description}
        </p>
        {item.location && (
          <p>
            <b>Location:</b> {item.location.city}, {item.location.regionOrState}
            , {item.location.country}
          </p>
        )}
      </div>
    </div>
  );
};

export default ItemCard;
