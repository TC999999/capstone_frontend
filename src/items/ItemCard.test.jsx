import { render } from "@testing-library/react";
import ItemCard from "./ItemCard.jsx";
import { TEST_ITEM } from "../_testCommons.js";
import { MemoryRouter } from "react-router-dom";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <ItemCard item={TEST_ITEM} />
    </MemoryRouter>
  );
});

it("should match snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <ItemCard item={TEST_ITEM} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("should have item-card", function () {
  const { container } = render(
    <MemoryRouter>
      <ItemCard item={TEST_ITEM} />
    </MemoryRouter>
  );
  const item_card = container.querySelector(".item-card");
  expect(item_card).toBeInTheDocument();
});
