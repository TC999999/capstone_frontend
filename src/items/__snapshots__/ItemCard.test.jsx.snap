// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="item-card"
  >
    <div
      class="item-image-name item-info"
    >
      <h3>
        <a
          href="/items/1"
        >
          i1
        </a>
      </h3>
      <img
        height="200"
        src="test_url_1"
        width="200"
      />
    </div>
    <div
      class="additional-item-info item-info"
    >
      <p>
        <b>
          Price:
        </b>
         $100
      </p>
      <p>
        <b>
          Seller:
        </b>
         
        <a
          href="/users/u1"
        >
          u1
        </a>
      </p>
      <p>
        <b>
          Condition:
        </b>
         great
      </p>
      <p>
        <b>
          Description:
        </b>
         test item
      </p>
      <p>
        <b>
          Location:
        </b>
         city1, state1, country1
      </p>
    </div>
  </div>
</DocumentFragment>
`;
